src/ripple/proto/org/xrpl/rpc/v1/common.proto

```diff
@@ -0,0 +1,451 @@
+syntax = "proto3";
+
+package org.xrpl.rpc.v1;
+option java_package = "org.xrpl.rpc.v1";
+option java_multiple_files = true;
+
+import "org/xrpl/rpc/v1/amount.proto";
+import "org/xrpl/rpc/v1/account.proto";
+
+// These fields are used in many different messsage types. They can be present
+// in one or more transactions, as well as metadata of one or more transactions.
+// Each is defined as it's own message type with a single field "value", to
+// ensure the field is the correct type everywhere it's used
+
+message ClearFlag
```

Comment:
Talked about this in a bit in another comment. I came up with this pattern myself, after I continuously had to check the type of the same field when writing the protobuf messages, as well as a couple times of getting the type wrong and having to track down the issue. This made my life so much easier writing the protobuf messages, and creates a more robust interface definition, that is easier to extend. And this pattern more correctly models what's actually going on with the API: DeliverMin is the same type everywhere it's used, which is intended, so giving it its own message enforces that invariant.
Also, for types that are actually scalars (CurrencyAmount is not a scalar, but it's own message type), wrapping like this prevents default initialization. Default initialization of these fields is a pain to deal with, because sometimes the default value actually means something, and there's no way to tell if the client actually set that value or not.
I don't think it makes the code more complicated in gRPCHelpers actually. Those functions are templated, so we ignore the type difference. In fact, off the top of my head, the types of these fields is hardly ever actually spelled out explicitly in the c++ code.

Reply:
@seelabs For what its worth, I would be open to "unwrapping" message types whose `value` field is another message type, as opposed to a scalar. Meaning instead of having `SendMax` and `DeliverMin` be separate types, I would just use `CurrencyAmount` for the type of both of those fields. The value won't be default initialized. The scalars I would still want to wrap, to avoid the default initialization at the very least.

Reply:
@cjcobb23 I'm fine with whatever you decide. I think of SendMax and DeliverMin as the same type, so I vote for that, but I don't think there's a huge difference if we make them the same type.